src/linker/Linker/LinkerAttributesInformation.cs

```diff
@@ -54,12 +57,17 @@ public static LinkerAttributesInformation Create (LinkContext context, ICustomAt
 					continue;
 
 				if (cache == null)
-					cache = new Dictionary<Type, List<Attribute>> ();
+					cache = new List<Tuple<Type, List<Attribute>>> ();
 
 				Type attributeValueType = attributeValue.GetType ();
-				if (!cache.TryGetValue (attributeValueType, out var attributeList)) {
+
+				var found = cache.Find (finder (attributeValueType));
+				List<Attribute> attributeList;
+				if (found == null || found == default) {
 					attributeList = new List<Attribute> ();
-					cache.Add (attributeValueType, attributeList);
+					cache.Add (Tuple.Create (attributeValueType, attributeList));
```

Comment:
With the changed type declaration above, this can be simply (and yes, this does unfortunately require double parenthesis):
```suggestion
					cache.Add ((attributeValueType, attributeList));
```